### PR TITLE
MM-10388: improved loadtest results

### PR DESCRIPTION
**Pull request by lieut-data**
This PR effects the following changes:
* Remove support for the termui/`ShowUI` stuff, as we're generally running this programmatically.
* Switch to structured logging via `mlog`, directly from `mattermost-server`.
* Dump results as JSON via structured logging
* Introduce `ltparse`, a tool for consuming loadtest log output. It looks for specially tagged logs identifying results, and supports `--text`, `--markdown`, `--aggregate` and `--baseline` flags:

I've included a sample (local) load test comparing a "baseline" with a current run. I'm having trouble using `ltops` with AWS to get a clean run, but it should be a straightforward matter of piping the results from `ltops` to `ltparse` within mattermod to get the results below.

Future ideas include:
* `--html` output that includes full logs and shows off more metrics
* clear pass/fail semantics (perhaps using :white_check_mark: and :x:) per route statistic

### Score: 15.31
The score is the the average of the mean reponse times below.

#### /channels/CID/members
| Metric | Baseline | Actual | Delta |
| --- | --- | --- | --- |
| Hits | - | 1 | - |
| Error Rate | - | 0.00% | - |
| Max Response Time | - | 3ms | - |
| Min Response Time | - | 3ms | - |
| Mean Response Time | - | 3.00ms | - |
| Median Response Time | - | 3.00ms | - |
| Inter Quartile Range | - | 0 | - |
#### /files
| Metric | Baseline | Actual | Delta | Delta % |
| --- | --- | --- | --- | --- |
| Hits | 10 | 10 | 0 | 0%
| Error Rate | 0.00% | 0.00% | 0% | 0% |
| Max Response Time | 26ms | 25ms | -1.00ms | -3.85% |
| Min Response Time | 3ms | 3ms | 0ms | 0% |
| Mean Response Time | 9.50ms | 9.80ms | +0.30ms | +3.16% |
| Median Response Time | 5.00ms | 7.00ms | +2.00ms | +40.00% |
| Inter Quartile Range | 8 | 5 | -3.00ms | -37.50% |
#### /posts
| Metric | Baseline | Actual | Delta | Delta % |
| --- | --- | --- | --- | --- |
| Hits | 469 | 246 | -223 | -47.55%
| Error Rate | 0.00% | 0.00% | 0% | 0% |
| Max Response Time | 40ms | 36ms | -4.00ms | -10.00% |
| Min Response Time | 11ms | 11ms | 0ms | 0% |
| Mean Response Time | 14.19ms | 14.79ms | +0.60ms | +4.25% |
| Median Response Time | 14.00ms | 14.00ms | 0ms | 0% |
| Inter Quartile Range | 2 | 2 | 0ms | 0% |
#### /users/UID/active
| Metric | Baseline | Actual | Delta | Delta % |
| --- | --- | --- | --- | --- |
| Hits | 5 | 5 | 0 | 0%
| Error Rate | 0.00% | 0.00% | 0% | 0% |
| Max Response Time | 19ms | 11ms | -8.00ms | -42.11% |
| Min Response Time | 8ms | 8ms | 0ms | 0% |
| Mean Response Time | 13.00ms | 9.60ms | -3.40ms | -26.15% |
| Median Response Time | 12.00ms | 10.00ms | -2.00ms | -16.67% |
| Inter Quartile Range | 9.5 | 2 | -7.50ms | -78.95% |
#### /users/email/UID
| Metric | Baseline | Actual | Delta | Delta % |
| --- | --- | --- | --- | --- |
| Hits | 5 | 5 | 0 | 0%
| Error Rate | 0.00% | 0.00% | 0% | 0% |
| Max Response Time | 7ms | 5ms | -2.00ms | -28.57% |
| Min Response Time | 2ms | 2ms | 0ms | 0% |
| Mean Response Time | 3.60ms | 3.20ms | -0.40ms | -11.11% |
| Median Response Time | 3.00ms | 3.00ms | 0ms | 0% |
| Inter Quartile Range | 2.5 | 1.5 | -1.00ms | -40.00% |
#### /users/status/ids
| Metric | Baseline | Actual | Delta | Delta % |
| --- | --- | --- | --- | --- |
| Hits | 4 | 2 | -2 | -50.00%
| Error Rate | 0.00% | 0.00% | 0% | 0% |
| Max Response Time | 3ms | 2ms | -1.00ms | -33.33% |
| Min Response Time | 1ms | 1ms | 0ms | 0% |
| Mean Response Time | 2.00ms | 1.50ms | -0.50ms | -25.00% |
| Median Response Time | 2.00ms | 1.50ms | -0.50ms | -25.00% |
| Inter Quartile Range | 1 | 1 | 0ms | 0% |

### CODE REVIEW

**Comment by lieut-data** [comment]
Note that it may be worth reviewing the commits independently, if you want to tease apart the switch to `mlog` from the rest of the changes. I'd be happy to break apart the PRs too if it were to help.

**Review by crspeller — APPROVED** [review]
Looks great!

**Comment by jasonblais** [comment]
Awesome!! One question: Was the above test a "pass" or a "failure"? Why? Asking as I'd love to see something like

> clear pass/fail semantics (perhaps using ✅ and ❌) per route statistic

to make it super obvious. Or perhaps a link to docs that (briefly) explains this output.

File: loadtest/run.go [hunk 1]
```diff
@@ -196,71 +178,63 @@ func RunTest(test *TestRun) error {
...
+	// 	mlog.Info("Sending results to mm server.")
+	// 	sendResultsToMMServer(
+	// 		cfg.ResultsConfiguration.ResultsServerURL,
+	// 		cfg.ResultsConfiguration.ResultsUsername,
+	// 		cfg.ResultsConfiguration.ResultsPassword,
+	// 		cfg.ResultsConfiguration.ResultsChannelId,
+	// 		cfg.ResultsConfiguration.CustomReportText,
+	// 		files,
+	// 	)
+	// }
```
Comment (jwilander) [inline loadtest/run.go:235]: Purposefully left in or should this be removed?

**Review by jwilander — APPROVED** [review]
Nice! Just one non-blocking comment

**Comment by crspeller** [comment]
Going to merge this since your out. (and I want this) Feel free to address comments with a patch.

**Pull request merged**